packages/database/src/modelClasses/Task.js

```diff
@@ -44,28 +103,89 @@ export class TaskRecord extends DatabaseRecord {
     return this.otherModels.survey.findById(this.survey_id);
   }
 
+  /**
+   * @description Get all comments for the task
+   * @returns {Promise<TaskCommentRecord[]>}
+   */
+
   async comments() {
     return this.otherModels.taskComment.find({ task_id: this.id });
   }
 
+  /**
+   * @description Get all user comments for the task
+   * @returns {Promise<TaskCommentRecord[]>}
+   */
   async userComments() {
     return this.otherModels.taskComment.find({ task_id: this.id, type: 'user' });
   }
 
+  /**
+   * @description Get all system comments for the task
+   * @returns {Promise<TaskCommentRecord[]>}
+   */
+
   async systemComments() {
     return this.otherModels.taskComment.find({ task_id: this.id, type: 'system' });
   }
+  /**
+   * @description Add a comment to the task. Handles linking the comment to the task and user, and setting the comment type
+   *
+   * @param {string} message
+   * @param {string} userId
+   * @param {string} type
+   *
+   */
 
-  async addComment(message, userId, type = 'user') {
+  async addComment(message, userId, type) {
     const user = await this.otherModels.user.findById(userId);
     return this.otherModels.taskComment.create({
-      message,
       task_id: this.id,
+      message,
+      type,
       user_id: userId,
       user_name: user.full_name,
-      type,
     });
   }
+
+  /**
+   * @description Add system comments when a task is updated. This is used to automatically add comments when certain fields are updated, e.g. due date, assignee, etc.
+   *
+   * @param {object} updatedFields
+   * @param {string} userId
+   */
+
+  async addSystemCommentsOnUpdate(updatedFields, userId) {
```

Comment:
Again I'm loving the object orientated approach here. I think that this logic is a perfect candidate for a unit test. We could add a Task model test in the database package or just update the EditTask tests in the central server with some extra tests.